Modules/Rat_Board.py

```diff
@@ -0,0 +1,197 @@
+"""
+Rat_Board.py - Rescue board
+
+Handles Tracking individual Rescues from creation to completion
+
+Copyright (c) 2018 The Fuel Rats Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.md
+
+This module is built on top of the Pydle system.
+"""
+import logging
+from uuid import UUID
+
+from Modules.rat_rescue import Rescue
+from config import Logging
+
+LOG = logging.getLogger(f"{Logging.base_logger}.{__name__}")
+
+
+class RescueBoardException(BaseException):
+    """
+    Generic Rescue Board Exception
+    """
+
+    def __init__(self, *args: object) -> None:
+        LOG.error(f"Rescueboard Exception raised!\nargs = {args}")
+        super().__init__(*args)
+
+
+class RescueNotFoundException(RescueBoardException):
+    """
+    Exception raised when a Rescue was not found
+    """
+    pass
+
+
+class RescueNotChangedException(RescueBoardException):
+    """
+    Exception raised when there was no net change to a Rescue.
+    """
+    pass
+
+
+class IndexNotFreeError(RescueBoardException):
+    """ Exception raised when someone attempts to attach a Rescue to a board index in-use"""
+    pass
+
+
+class RatBoard(object):
+    """
+    A rescue board
+    """
+
+    def __init__(self, handler=None):
+        """
+        Create a new Rescue Board.
+
+        Args:
+            handler (): WS API handler
+        """
+        self.handler = handler
+        self._rescues = {}
+        """Rescue objects tracked by this board"""
+
+    def find_by_index(self, index: int) -> Rescue or None:
+        """
+        Searches for and returns a Rescue at a given `index` position, should it exist
+        Args:
+            index (int): case number to return
+
+        Returns:
+            Rescue: rescue found
+            None: no rescue found
+
+        """
+        try:
+            found = self._rescues[index]
+        except KeyError:
+            # the key doesn't exist, therefore no rescue at that index.
+            pass
+        else:
+            # we found something
+            return found
+
+    def find_by_name(self, client: str) -> Rescue or None:
+        """
+        Searches for and returns a Rescue for a given client, should it exist.
+
+        Returns:
+            Rescue: found rescue
+            None:   no such rescue
+        """
+        for index, rescue in self._rescues.items():
```

Comment:
When you don't need the keys, dict.values() will provide you with a set (ish) representation of the values.

Reply:
*nods* til